### DISCUSSION TITLE
Utilizing Existing Knowledge Base

### DISCUSSION THREAD

**Discussion by sharva89** [MacPaw/OpenAI#107] [Q&A]
Let us assume, I have existing knowledge base in the form of pdf or descriptions. How do I ask OpenAI to respond from that ?

**Comment by cdillard** [comment]
Yeah you can use PDF with the assistants API in my WIP [PR](https://github.com/MacPaw/OpenAI/pull/140) for the assistants API here.
Just make sure you select a PDF and select "Retrieval" when creating your assistant, then the assistant will automatically use RAG when required to look up information in your uploaded PDF.
